enhancements/network/ovn-observability-api.md

```diff
@@ -0,0 +1,262 @@
+Cluster-scoped features, like (Baseline)AdminNetworkPolicy, can only be additionally filtered based on name. Label selectors
+can't be used, as observability module doesn't watch any objects.
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+As long as netobserv can be installed on the management or hosted cluster, integration should work fine.
+If we need to make it work for shared-cluster netobserv deployment, more work.testing will be needed.
+Netobserv team confirmed that separate installations for management and hosted clusters are supported as a result of https://issues.redhat.com/browse/NETOBSERV-1196.
+
+For debuggability mode without netobserv integration, everything should work separately on every cluster with its own ovn-kubernetes instance.
+
+#### Standalone Clusters
+#### Single-node Deployments or MicroShift
+
+SNO should be supported with netobserv, Microshift is not.
+Microshift most likely will not be supported even without netobserv, as observability requires new features to be enabled
+and adds overhead to the cluster.
+
+### Implementation Details/Notes/Constraints
+
+Filtering for debugging use case may be tricky as explained in "Observability filtering for debugging" section.
+But it is important to implement, as OVN implementation works in a way, where performance optimizations can only be
+applied when there is only 1 `ObservabilityConfig` for a given feature. That means, adding a second `ObservabilityConfig`
+for a feature will have a bigger performance impact than creating a new one.
+
+### Risks and Mitigations
+
+The biggest risk of enabling observability is performance overhead.
+This will be mitigated by scale-testing and sampling percentage configuration.
+
+### Drawbacks
+
+- e2e testing upstream is not possible until github runners allow using kernel 6.11
+
+## Test Plan
+
+- e2e downstream
+- perf/scale testing for dataplane and control plane as a GA plan
+- netobserv testing for correct GUI representation
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+None
+
+### Tech Preview -> GA
+
+- More testing (upgrade, downgrade)
+- Available by default, but no overhead until observability config CR is created
+- Run perf/scale tests and document related metrics to track perf overhead, write down any limitations/recommendations
+based on the workload and cluster size
+- User facing documentation created in openshift-docs and netobserv docs
+
+### Removing a deprecated feature
+
+## Upgrade / Downgrade Strategy
+
+No upgrade impact is expected, as this feature will only be enabled when a new CR is created.
+
+## Version Skew Strategy
+
+All component were designed to be backwards-compatible
+- old kernel + new OVS: TDB
+- old OVS + new OVN: TDB
+- old OVN + new ovn-k: rolled out at the same time, no version skew is expected
+- old kernel/OVS/ovn-kubernetes + new netobserv: TDB
```

Comment:
old kernel/OVS/ovn-kubernetes + new netobserv => netobserv can detect if it's running on older ocp version and warn when the users try to enable the feature.